### ISSUE TITLE
VM Scale Set Scale-In Properties update

### ISSUE THREAD

**Issue by jontreynes** [terraform-providers/terraform-provider-azurerm#4419]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add the new scale-in schema in 
https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_virtual_machine_scale_set.go as described here

https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate#scaleinpolicy

### New or Affected Resource(s)
resource_arm_virtual_machine_scale_set.go

* azurerm_virtual_machine_scale_set

### Potential Terraform Configuration

```hcl
#NA at this time
```

### References

* Links are above

**Comment by jontreynes** [comment]
@tombuildsstuff thoughts?

**Comment by tombuildsstuff** [comment]
hey @jontreynes 

Thanks for opening this issue :)

I've spent the last few weeks working on the new Virtual Machine Scale Set resources we outlined in #2807 - as a part of this I've been going through the open VMSS issues to ensure everything's covered. Taking a look at the Azure API it appears there's two potential bits to this:

1. Configuring this on the VM Scale Set itself (which you've requested here) - which unfortunately currently isn't exposed in the Azure SDK
2. Configuring this on each instance within the VM Scale Set - which /is/ supported in the Go SDK but doesn't really make sense for us to support in Terraform (since this requires the Instance ID where the instances could be cycled outside of TF).

As such unfortunately this is blocked on support in the Azure SDK for Go - for the moment I'm going to assign this to the `Blocked` milestone and assign `upstream-microsoft` (which we use to track these issues) - but when support becomes available in the Go SDK it should be possible to add support for this 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
👋 

At this point in time the older `azurerm_virtual_machine_scale_set` resource has been superseded by the `azurerm_linux_virtual_machine_scale_set` and `azurerm_windows_virtual_machine_scale_set` resources. Since the older `azurerm_virtual_machine_scale_set` resource has fallen out of sync from the behaviour of the Azure API, to avoid unintentionally breaking users relying on this behaviour the older resources are feature-frozen.

As this resource is feature-frozen to maintain compatibility, I'm going to close this issue for the moment - but I'd recommend instead checking out the `azurerm_linux_virtual_machine_scale_set`/`azurerm_windows_virtual_machine_scale_set` resources, which better represents how the Azure API behaves for this resources and adds support for this functionality.

Thanks!
